<launch>
    
    <node pkg="nodelet" type="nodelet" name="pcl_manager" args="manager" output="screen" />

    <!-- Run a VoxelGrid filter to clean NaNs and downsample the data -->
    <node pkg="nodelet" type="nodelet" name="voxel_grid" args="load pcl/VoxelGrid pcl_manager" output="screen">
        <remap from="~input" to="/bumblebee2/points2" />
        <rosparam>
            filter_field_name: y
            filter_limit_min: -10.0
            filter_limit_max: 10.0
            filter_limit_negative: False
            leaf_size: 0.1
        </rosparam>
    </node>

    <node pkg="nodelet" type="nodelet" name="statistical_outlier" args="load pcl/StatisticalOutlierRemoval pcl_manager" output="screen">
        <remap from="~input" to="/voxel_grid/output" />
        <rosparam>
            mean_k: 10
            stddev: 0.5
        </rosparam>
    </node>
</launch>